xonsh/proc.py

```diff
@@ -541,14 +542,38 @@ def _wcode_to_popen(code):
                                    "stdin_redirect",
                                    "stdout_redirect",
                                    "stderr_redirect",
-                                   "timestamp"])
+                                   "timestamp",
+                                   "executed_cmd"])
 
 class CompletedCommand(_CCTuple):
     """Represents a completed subprocess-mode command."""
 
     def __bool__(self):
         return self.returncode == 0
 
+    def __iter__(self):
+        stdout = self.stdout
+        start = 0;
+        end = 0
+        while end != -1:
+            end = stdout.find('\n', start)
+            if end == -1:                # no newlines, but possibly more text
+                snippet = stdout[start:]
+            elif stdout[end-1] == '\r':  # newline, check for CR
+                snippet = stdout[start:end-1]
+            else:                        # newline, no CR
+                snippet = stdout[start:end]
+            if snippet or not (end == -1):
```

Comment:
Ah, that's quite nice.  Also retains the ability to check whether `\n` occurs at all, which is an important piece here.

How did I not know about this function (`parition`) until just now?!